lib/parameter_substitution/expression.rb

```diff
@@ -33,6 +36,17 @@ def warnings
       unknown_parameters_message
```

Comment:
I wonder if this method needs a more specific name now that we're grouping it right next to `parameter_and_method_warnings`. The `web` code implicitly knows that they are `legacy_errors`. I don't really have an idea for what to call it, but it is a little confusing now.